TruthTables/Workbook_TruthTables.ipynb

```diff
@@ -468,6 +468,138 @@
+    "\n",
+    "* [IntAsBoolArray](https://learn.microsoft.com/qsharp/api/qsharp/microsoft.quantum.convert.intasboolarray) function will convert the truth table stored as an integer into an array of individual bits; the bit at index `ind` will be the value of the function for the input assignment `ind`.\n",
+    "* [Enumerated](https://learn.microsoft.com/qsharp/api/qsharp/microsoft.quantum.arrays.enumerated) function will convert the array of values into an array of tuples, attaching the index `ind` to each bit.\n",
+    "* [Filtered](https://learn.microsoft.com/qsharp/api/qsharp/microsoft.quantum.arrays.filtered) function will filter out the tuples based on the value of the bit, keeping only the ones for which it is true.\n",
+    "* Finally, [Mapped](https://learn.microsoft.com/qsharp/api/qsharp/microsoft.quantum.arrays.mapped) function will keep only the indices of the remaining tuples.\n",
+    "* [Fst](https://learn.microsoft.com/qsharp/api/qsharp/microsoft.quantum.canon.fst) and [Snd](https://learn.microsoft.com/qsharp/api/qsharp/microsoft.quantum.canon.snd) functions return the first and the second elements of the input tuple, respectively."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%kata T09_AllMinterms\n",
+    "\n",
+    "open Microsoft.Quantum.Arrays;\n",
+    "open Microsoft.Quantum.Convert;\n",
+    "open Quantum.Kata.TruthTables;\n",
+    "\n",
+    "function AllMinterms (tt : TruthTable) : Int[] {\n",
+    "    let bits = IntAsBoolArray(tt::bits, 2^tt::numVars);\n",
+    "    let indexedBits = Enumerated(bits);\n",
+    "    let filteredTuples = Filtered(Snd, indexedBits);\n",
+    "    return Mapped(Fst, filteredTuples);\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "[Return to task 9 of the Truth Tables kata.](./TruthTables.ipynb#Task-9.-Find-all-true-input-assignments-in-a-truth-table)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Task 10. Apply the truth table as a quantum operation\n",
+    "**Goal:** \n",
+    "Apply the X operation on the target qubit, if and only if\n",
+    "the classical state of the controls is a minterm of the truth table."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Solution\n",
+    "\n",
+    "A typical controlled gate (implemented by the [ControlledOnInt](https://learn.microsoft.com/qsharp/api/qsharp/microsoft.quantum.canon.controlledonint) function) applies the gate if the classical state of the control qubits matches the given integer. \n",
+    "In our case, we need to apply the X gate if the classical state matches any one of the minterms; we can implement this by applying a series of controlled X gates, one per minterm. Since all minterms are different, the X gate will never end up being applied two or more times."
```

Comment:
Maybe it's worth to reference https://learn.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.synthesis.applyxcontrolledontruthtable here.